tests/python_package_test/test_basic.py

```diff
@@ -332,6 +332,133 @@ def check_asserts(data):
     check_asserts(lgb_data)
 
 
+def test_category_encoding(tmp_path):
+
+    def test_category_encoding_inner(tmp_path, X_train, X_test, y_train, y_test, params, model_prefix):
+        # checks that category_encoders works for Dataset constructor
+        category_encoders_str = "target,count,target:0.5,raw"
+        train_data_1 = lgb.Dataset(X_train, label=y_train, category_encoders=category_encoders_str)
+        valid_data_1 = train_data_1.create_valid(X_test, label=y_test)
+
+        categorical_feature = [fidx for fidx in range(X_train.shape[1] // 2)]
+        expected_num_features = X_train.shape[1] + 3 * len(categorical_feature)
+
+        params.update({"categorical_feature": categorical_feature})
+        booster_1 = lgb.train(params, train_data_1, valid_sets=[valid_data_1], valid_names=["valid_data"])
```

Comment:
Please check valid data creation via multiple ways: `create_valid`, `set_reference`, `reference` param in constructor, `add_valid`.

Reply:
Resolved via a25f323. All 4 methods are tested in the `test_category_encoding_inner` function.